lib/ansible/modules/database/mysql/mysql_db.py

```diff
@@ -346,81 +383,74 @@ def main():
     changed = False
     if not os.path.exists(config_file):
         config_file = None
-    if db_exists(cursor, db):
-        if state == "absent":
-            if module.check_mode:
-                module.exit_json(changed=True, db=db_name)
+
+    existence_list = []
+    non_existence_list = []
+    for each_database in db:
+        if db_exists(cursor, [each_database]):
+            existence_list.append(each_database)
+        else:
+            non_existence_list.append(each_database)
+
+    if state == "absent":
+        if module.check_mode and not db_exists(cursor, db):
+            module.exit_json(changed=False, db=db_name, db_list=db)
+        elif module.check_mode and db_exists(cursor, db):
```

Comment:
In case check mode is enabled, you are making the same call to `db_exists` **twice** (in case the first one returns `False`). That's not exactly efficient.

How about something like:
```
if module.check_mode:
    module.exit_json(changed=db_exists(cursor, db), db=db_name, db_list=db)
```

Reply:
Also, why do you need to call `db_exists` again at all? You already have `existence_list` and `existence_list`, that should be enough for the rest of the module.

Reply:
Finally, `db_exists` checks whether **all** DBs in the list exist. This doesn't really help here. You really need to work with `existence_list` and `existence_list`.

Reply:
Fixing it